pandas/core/indexes/interval.py

```diff
@@ -526,6 +522,10 @@ def is_overlapping(self) -> bool:
         # GH 23309
         return self._engine.is_overlapping
 
+    def holds_integer(self):
+        return self.dtype.subtype.kind not in ["m", "M"]
```

Comment:
this is ok, but doesn't follow the other patterns of inference, where we use infer_dtype. but for known dtypes this is (sligthly) cheaper.

Reply:
see holds_integer in the indexes/base.py

Reply:
@jschendel the Float64Index behavior was changed in #31329 so s_fi.at[1] returns "b".  This was done to make .at behavior match .loc behavior.

Deferring to self.left or self.right seems reasonable, but we may need something other than holds_integer.  Most of the places where it is used are de-facto used like it is here, but that doesn't match the docstring description.

Reply:
this seems fine for now, @jbrockmendel if you can make a note to re-visit later